src/utils/mergeDeep.js

```diff
@@ -0,0 +1,28 @@
+const isObject = (obj) => obj && typeof obj === 'object' && !Array.isArray(obj);
+
+/**
+ * Performs a deep merge of objects and returns new object.
+ *
+ * @param {...object} objects
+ * @returns {object}
+ */
+export const mergeDeep = (...objects) => objects.reduce((prev, obj) => {
+  if (obj == null) {
+    return prev;
+  }
+
+  const newObject = { ...prev };
+
+  Object.keys(obj).forEach((key) => {
+    const pVal = prev[key];
+    const oVal = obj[key];
```

Comment:
While `obj` could be acceptable, I have no idea what `pVal` or `oVal` means. Can we come up with better names please? 🙂

Reply:
+1